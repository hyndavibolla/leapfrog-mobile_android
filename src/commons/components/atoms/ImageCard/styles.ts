import { css } from '@emotion/native';

export const styles = {
  imageContainer: css`
    padding: 4px 0;
  `,
  backgroundContainer: css`
    border-radius: 8px;
    height: 100%;
    overflow: hidden;
    width: 100%;
  `
};
