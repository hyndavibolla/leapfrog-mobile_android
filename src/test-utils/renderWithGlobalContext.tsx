import React, { ReactElement } from 'react';
import { render } from '@testing-library/react-native';

import { getInitialState, GlobalProvider } from '../state-mgmt/GlobalState';
import { Deps, IGlobalState, IReducer } from '../models/general';
import { getMockDeps } from './getMockDeps';
import { getMockReducer } from './getMockReducer';

export const renderWithGlobalContext = (
  children: ReactElement,
  deps: Deps = getMockDeps(),
  initState: IGlobalState = getInitialState(),
  combinedReducers: IReducer<IGlobalState, any> = getMockReducer()
) => {
  const renderAPI = render(
    <GlobalProvider deps={deps} initState={initState} combinedReducers={combinedReducers}>
      {children}
    </GlobalProvider>
  );
  return { ...renderAPI, mockReducer: combinedReducers, deps };
};
