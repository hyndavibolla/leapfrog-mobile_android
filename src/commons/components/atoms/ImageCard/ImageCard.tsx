import React, { memo } from 'react';
import { View } from 'react-native';
import { ResizeMode } from 'react-native-fast-image';

import { ImageBackgroundWithFallback } from '_components/ImageBackgroundWithFallback';
import { CONTAINER_STYLE } from '_constants';
import { styles } from './styles';

export interface Props {
  imageUrl: string;
  resizeMode?: ResizeMode;
}

const ImageCard = ({ imageUrl, resizeMode = 'cover' }: Props) => {
  return (
    <View style={styles.imageContainer}>
      <ImageBackgroundWithFallback
        containerStyle={[CONTAINER_STYLE.shadow, styles.backgroundContainer]}
        source={{ uri: imageUrl }}
        fallbackSource={require('_assets/shared/categoryBackgroundFallback.png')}
        resizeMode={resizeMode}
      />
    </View>
  );
};

export default memo(ImageCard);
