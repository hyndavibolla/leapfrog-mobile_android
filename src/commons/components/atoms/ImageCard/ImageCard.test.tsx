import React from 'react';
import { act } from 'react-test-renderer';

import { getMission_1 } from '_test_utils/entities';
import { renderWithGlobalContext } from '_test_utils/renderWithGlobalContext';
import { wait } from '_utils/wait';
import ImageCard, { Props } from './ImageCard';

describe('App Common WebView ', () => {
  let props: Props;

  beforeEach(() => {
    props = {
      imageUrl: getMission_1().image
    };
  });

  it('should render', async () => {
    const { toJSON } = renderWithGlobalContext(<ImageCard {...props} />);
    await act(async () => await wait(0));
    expect(toJSON()).toMatchSnapshot();
  });
});
