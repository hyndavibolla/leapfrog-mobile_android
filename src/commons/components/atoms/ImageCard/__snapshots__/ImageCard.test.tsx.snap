// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Common WebView  should render 1`] = `
<View
  style={
    Object {
      "paddingBottom": 4,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 4,
    }
  }
>
  <View
    accessibilityLabel="image-background-container"
    accessible={true}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "elevation": 3,
          "shadowColor": "#262C2F",
          "shadowOffset": Object {
            "height": 4,
            "width": 0,
          },
          "shadowOpacity": 0.1,
        },
        Object {
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "height": "100%",
          "overflow": "hidden",
          "width": "100%",
        },
      ]
    }
    testID="image-background-container"
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": "100%",
              "width": "100%",
            },
            undefined,
          ],
        ]
      }
    >
      <FastImageView
        accessibilityLabel="image-background-image"
        accessible={true}
        onFastImageError={[Function]}
        onFastImageLoadEnd={[Function]}
        onFastImageLoadStart={[Function]}
        resizeMode="cover"
        source={
          Object {
            "uri": "https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcTVkMt3KFinkf_ehnCq5yk2DLZBls6GYFZLeA&usqp=CAU",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        testID="image-background-image"
      />
    </View>
  </View>
</View>
`;
